helm/memcached-operator/deploy/role.yaml

```diff
@@ -7,37 +7,62 @@ rules:
 - apiGroups:
   - ""
   resources:
-  - pods
-  - services
-  - endpoints
-  - persistentvolumeclaims
-  - events
+  - namespaces
+  verbs:
+  - get
+- apiGroups:
+  - ""
+  resources:
   - configmaps
   - secrets
   verbs:
   - '*'
 - apiGroups:
   - ""
   resources:
-  - namespaces
+  - services
   verbs:
-  - get
+  - '*'
 - apiGroups:
   - apps
   resources:
   - deployments
-  - daemonsets
-  - replicasets
   - statefulsets
   verbs:
   - '*'
+- apiGroups:
+  - policy
+  resources:
+  - poddisruptionbudgets
+  verbs:
+  - '*'
 - apiGroups:
   - monitoring.coreos.com
   resources:
   - servicemonitors
   verbs:
   - get
   - create
+- apiGroups:
+  - apps
+  resourceNames:
+  - memcached-operator
+  resources:
+  - deployments/finalizers
+  verbs:
+  - update
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  verbs:
+  - get
+- apiGroups:
+  - apps
+  resources:
+  - replicasets
```

Comment:
Although we also give full permissions `deployments` in line 29, I think we should add it here as well, just to be consistent with operator-framework/operator-sdk#2090.

We can leave the `deployments` in line 29, because this chart can use either `StatefulSet` (default) or `Deployment` (by setting `.Values.kind` to `Deployment`)